web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -267,21 +284,52 @@ export function UnifiedResources(props: UnifiedResourcesProps) {
     updateUnifiedResourcesPreferences({ defaultTab: value });
   };
 
-  const $pinAllButton = (
-    <ButtonBorder
-      onClick={() => handlePinSelected(shouldUnpin)}
-      textTransform="none"
-      disabled={pinning.kind === 'not-supported'}
-      css={`
-        border: none;
-        color: ${props => props.theme.colors.brand};
-      `}
-    >
-      <PushPin color="brand" size={16} mr={2} />
-      {shouldUnpin ? 'Unpin ' : 'Pin '}
-      Selected
-    </ButtonBorder>
-  );
+  const resourcesForAction = useMemo(() => {
+    return resources
+      .filter(({ resource }) =>
+        selectedResources.includes(generateUnifiedResourceKey(resource))
+      )
+      .map(({ resource }) => ({
+        resource: resource,
+        unifiedResourceId: generateUnifiedResourceKey(resource),
+      }));
+  }, [selectedResources, resources]);
+
+  const pinAllButton = () => {
+    // if pining not supported, we don't need to build
+    // the component below
+    if (pinning.kind === 'hidden') {
+      return null;
+    }
+
+    const $pinButton = (
+      <ButtonBorder
+        textTransform="none"
+        onClick={() => handlePinSelected(shouldUnpin)}
+        disabled={
+          pinning.kind === 'not-supported' ||
+          updatePinnedResourcesAttempt.status === 'processing'
+        }
+        css={`
+          border: none;
+          color: ${props => props.theme.colors.brand};
+        `}
+      >
+        <PushPin color="brand" size="small" mr={2} />
+        {shouldUnpin ? 'Unpin Selected' : 'Pin Selected'}
+      </ButtonBorder>
+    );
+
+    if (pinning.kind !== 'not-supported') {
```

Comment:
IMO it is easier to understand:
```suggestion
    if (pinning.kind === 'supported') {
````